### DISCUSSION TITLE
Speed issues - APIs and emails

### DISCUSSION THREAD

**Discussion by igoraugustynski** [swellstores/community#359] [Help] [answered]
Hello,
I'm still experiencing many issues with very slow response times on your APIs. Also, I've noticed that password reset emails take about 15-20 minutes to arrive, which is weird. Would upgrading to the Starter plan (17$/month) help with this issues? Are there any guarantees included in those plans?

**Comment by igoraugustynski** [comment]
Anyone?

**Comment by swell-dev-support** [comment]
hi @igoraugustynski, can you provide a few cURL calls of these slow calls? And which account are you using? Will investigate.

**Reply by igoraugustynski** [reply]
Hey there,
I've installed some Firebase Performance Monitoring probes on the FE. Will get back to You in about a week with a detailed analysis of the issue.
In the meantime, two questions remain:
- Would upgrading to the Starter plan get me faster API response times?
- What about the reset password emails?

**Reply by swell-dev-support** [reply]
Hi Igor,
1. I can look at the requests you're experiencing latency with if you can provide a cURL example. If you are connecting from EU, we are planning to launch an EU availability zone in the coming weeks.
2. If you provide me with your account username, I can look at logs for the email issue too.

**Reply by igoraugustynski** [reply]
Hello,
please find attached a screenshot of Firebase Performance Monitoring. As You can see the place-order call is by far too slow. 
1. I am connecting from the EU so that would be great. You still haven't answered my question though, would upgrading get me better response times?
2. The account is `parsifal-commerce`.

<img width="1307" alt="Screenshot 2023-09-11 at 09 58 39" src="https://github.com/swellstores/community/assets/94485687/99395c0e-56eb-48df-87b8-be221fcbf393">

**Reply by logeshswell** [reply]
> would upgrading get me better response times?

Not necessarily.

**Reply by logeshswell** [reply]
@swell-dev-support will check and update you on the root cause of this performance issue and assist you accordingly.

**Reply by igoraugustynski** [reply]
Okay, thank you.

**Reply by igoraugustynski** [reply]
It's been two weeks, any updates on this?

**Reply by mewa** [reply]
I'm seeing similarly terrible performance, especially when updating the cart.

**Comment by mewa** [comment]
Just to reiterate how bad the speed issues are: we're tracking the number of users that click "go to checkout" and the number of users that actually "begin checkout" (i.e. implemented as checkout page visit).

### Over 40% of users drop between clicking "Go To Checkout" and "Beginning Checkout"

This number is abnormally high and suggests that the site & checkout (i.e. the API) is **VERY SLOW**.

Had the response times been good these 2 numbers should be very close to one another.

**Comment by ericingram** [comment, marked as answer]
The team made some adjustments this week and we are seeing better response times across the board now.
